docs/source/posts/2023/2023-08-24-final-report-tvcastillod.rst

```diff
@@ -0,0 +1,175 @@
+
+
+Proposed Objectives
+-------------------
+
+- Implement a parallelized version of computer-generated billboards using geometry shaders for amplification.
+- Model the mathematical functions that express the geometry of ellipsoids glyphs and implement them using Ray Marching techniques.
+- Model the mathematical functions that express the geometry of ODF glyphs and implement them using Ray Marching techniques.
+- Use SDF properties and techniques to represent the uncertainty of dMRI reconstruction models.
+
+
+Objectives Completed
+--------------------
+
+Ellipsoid actor implemented with SDF
+************************************
+
+A first approach for tensor glyph generation has been made, using raymarching and SDF applied to a box. The current implementation with ``tensor_slicer`` requires a sphere with a specific number of vertices to be deformed based on this model, to get a higher resolution a sphere with more vertices is needed. Because the raymarching technique does not use polygonal meshes it is possible to define perfectly smooth surfaces and still obtain a fast rendering.
+
+Details of the implementation:
+
+- *Vertex shader pre-calculations*: Some minor calculations are done in the vertex shader. One corresponding to the eigenvalues constraining and min-max normalization to avoid incorrect visualizations when the difference between the eigenvalues is too large. And the other related to the tensor matrix calculation given by the diffusion tensor definition :math:`T = R^{−1}\Lambda R` where :math:`R` is a rotation matrix that transforms the standard basis onto the eigenvector basis, and :math:`\Lambda` is the diagonal matrix of eigenvalues [4]_.
+- *Ellipsoid SDF definition*: The definition of the SDF is done in the fragment shader inside the ``map`` function which is used later for the raymarching algorithm and the normals calculation. We define the SDF in a simpler way by transforming a sphere into an ellipsoid, considering that the SDF of a sphere is easily computed and the definition of a tensor gives us a linear transformation of a given geometry. Also, as scaling is not a rigid body transformation, we multiply the final result by a factor to compensate for the difference, which gave us the SDF of the ellipsoid defined as ``sdSphere(tensorMatrix * (position - centerMCVSOutput), scaleVSOutput*0.48) * scFactor``.
+- *Raymarching algorithm and lighting*: For the raymarching algorithm a small value of 20 was taken as the maximum distance since we apply the technique to each individual object and not all at the same time, additionally, we set the convergence precision to 0.001. We use the central differences method to compute the normals necessary for the scene’s illumination, besides the Blinn-Phong lighting technique which is high-quality and computationally cheap.
```

Comment:
`For the raymarching algorithm a small value` -> `For the raymarching algorithm, a small value`